runtime/src/iree/task/queue_test.cc

```diff
@@ -6,8 +6,26 @@
 
 #include "iree/task/queue.h"
 
+#include <thread>
+
 #include "iree/testing/gtest.h"
 
+// Like iree_task_queue_try_steal but retries until success.
+// This is used in this test as iree_task_queue_try_steal may (rarely) fail even
+// in simple tests that don't look particularly contended, see #15488.
+static iree_task_t* iree_task_queue_try_steal_until_success(
+    iree_task_queue_t* source_queue, iree_task_queue_t* target_queue,
+    iree_host_size_t max_tasks) {
+  while (true) {
+    iree_task_t* task =
+        iree_task_queue_try_steal(source_queue, target_queue, max_tasks);
+    if (task) {
+      return task;
+    }
+    std::this_thread::yield();
```

Comment:
use iree_thread_yield from iree/base/internal/threading.h instead

Reply:
done! (didn't know about that one! I saw `iree_processor_yield` static in `synchronization.c` but it's kind of different, as the name indicates. I'm actually not clear as to what the exact difference is, and so i dont understand why this is both a CPU instruction that userspace code can directly run (as in `iree_processor_yield`) and a OS syscall (as in `iree_thread_yield`) ?

Reply:
Self-answering, `iree_processor_yield` is just a hint that it's ok to deprioritize this thread, it doesn't immediately end the thread timeslice in the way a yield syscall does. So, the reason why `iree_processor_yield` is static in `synchronization.c` is that it is something you really only would use in the implementation of a spinlock.